api/handler/repo.go

```diff
@@ -2085,21 +2084,20 @@ func (h *RepoHandler) ServerlessUpdate(ctx *gin.Context) {
 		return
 	}
 
-	var req types.ModelRunReq
+	var req *types.DeployUpdateReq
 	if err := ctx.ShouldBindJSON(&req); err != nil {
 		slog.Error("Bad request format", "error", err, slog.Any("request.body", ctx.Request.Body))
 		httpbase.BadRequest(ctx, err.Error())
 		return
 	}
 
-	if req.Revision == "" {
-		req.Revision = "main" // default repo branch
-	}
-
-	if req.MinReplica < 0 || req.MaxReplica < 0 || req.MinReplica > req.MaxReplica {
-		slog.Error("Bad request setting for replica", slog.Any("MinReplica", req.MinReplica), slog.Any("MaxReplica", req.MaxReplica))
-		httpbase.BadRequest(ctx, "Bad request setting for replica")
-		return
+	if req.MinReplica != nil && req.MaxReplica != nil {
+		err = Validate.Struct(req)
+		if err != nil {
+			slog.Error("Bad request setting for serverless", slog.Any("req", *req), slog.Any("err", err))
+			httpbase.BadRequest(ctx, "Bad request setting for serverless")
```

Comment:
please also send validation error to the client in response

Reply:
done